hack/teardown.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script tears down the installed Knative Eventing components from source for local development.
+
+set -e
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "$0")/../test/e2e-common.sh"
+
+read -a UNINSTALL_LIST < $(dirname "$0")/../tmp/uninstall_list.txt
```

Comment:
It appears that if the installation process was not completed using install.sh,  the file `tmp/uninstall_list.txt`, will be missing. 

So could you add some prompt to users to explain the reason why this may fail? e.g Why we need uninstall_list.txt or something like that

```
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
./hack/../test/../vendor/knative.dev/hack/library.sh: line 84: 1: unbound variable
./hack/../test/../vendor/knative.dev/hack/library.sh: line 93: 1: unbound variable
./hack/teardown.sh: line 26: ./hack/../tmp/uninstall_list.txt: No such file or directory
```

Reply:
Right, so for this we can check whether the file exists or not, if it doesn't we will explain that install.sh must have failed so this file wasn't created and could just ask them to delete the namespaces and crds manually.
Something like - 
```
if [[ -e $(dirname "$0")/../tmp/uninstall_list.txt ]]; then
    read -a UNINSTALL_LIST < $(dirname "$0")/../tmp/uninstall_list.txt
else
    echo "install.sh failed to create ./tmp/uninstall_list.txt. Delete namespaces and crds manually if created" && exit
fi
```

And add an explanation about uninstall_list.txt as a comment in the script itself.

Reply:
These two lines throw out the error - 
https://github.com/pawarpranav83/eventing/blob/bb5313df8dd5295573cbc4a08206af741edeeef0/vendor/knative.dev/hack/library.sh#L998-L1002

Reply:
I think we can ignore this error in this PR, and fix it in other PR.

Reply:
@pawarpranav83   Please pull the latest change and merge it to your current branch or close the PR and create a new one.

As I tested locally, I copied all your code changes to a new branch that is up-to-date with the latest upstream main branch, and no error is reported.

Reply:
Ig it depends on the branch name, but it doesn't affect the script actually since all the resources that were installed are deleted regardless of that error (that function isn't used so...).